adapter/internal/operator/controllers/dp/api_controller.go

```diff
@@ -194,8 +203,16 @@ func NewAPIController(mgr manager.Manager, operatorDataStore *synchronizer.Opera
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.0/pkg/reconcile
 func (apiReconciler *APIReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	if !readyToReconcile {
+		mutexForReconcile.Lock()
+		defer mutexForReconcile.Unlock()
+		if !readyToReconcile {
+			loggers.LoggerAPKOperator.Infof("Ignore reconciling for API %s as the startup API apply is pending.", req.NamespacedName.String())
+			apiReconciler.applyStartupAPIs()
```

Comment:
I feel like this will get called more than once. I guess we need another bool with mutex to call `applyStartupAPIs` once.

shall we introduce functions to read and write to these bools like:
```
func isReadyToReconcile() bool {
    mutexForReconcile.Lock()
    defer mutexForReconcile.Unlock()
    return readyToReconcile
}

func setReadyToReconcile(value bool) {
    mutexForReconcile.Lock()
    defer mutexForReconcile.Unlock()
    myBool = true
} 
```
so that we can use:

```
if !isReadyToReconcile() {
   if !isCalledApplyStartupAPIs() {
       apiReconciler.applyStartupAPIs()
   }
   return ctrl.Result{}, nil
}
```